### DISCUSSION TITLE
Test UPT-R50 on HICO-DET use python

### DISCUSSION THREAD

**Discussion by jasminethurder** [fredzzhang/upt#73] [Q&A] [answered]
I meet some problem when i try to Test UPT-R50 on HICO-DET. i use "python main.py --eval --resume checkpoints/upt-r50-hicodet.pt" according to the [launch_template.sh](https://github.com/fredzzhang/upt/blob/main/launch_template.sh), but in the end, mAP is zero.
![图片](https://user-images.githubusercontent.com/86194554/221955552-94dcd336-bdca-4ec7-ac3a-0369e5731695.png) 
Thank you very much!

**Comment by fredzzhang** [comment, marked as answer]
Hi @jasminethurder,

Could you try to do some visualisations by running `python inference.py --resume checkpoints/upt-r50-hicodet.pt --image-path ./assets/umbrella.jpeg --action 36` and see if you can get the same results as shown in the front page?

Fred.

**Reply by jasminethurder** [reply]
Yes, it works when i do the visualisations.
<img width="640" alt="图片" src="https://user-images.githubusercontent.com/86194554/222045106-23e91780-e803-4b05-a131-4ba059d105dc.png">

**Reply by jasminethurder** [reply]
i alse write some visualisations base on OpenCV, The model seems to run well, but it always returns 0mAP when i use "python main.py --eval --resume checkpoints/upt-r50-hicodet.pt" to test.
<img width="1116" alt="图片" src="https://user-images.githubusercontent.com/86194554/222046279-65bab5fc-66da-4129-b716-6cb7da136099.png">

Even if I reduce the IOU threshold from 0.5 to 0.0, the final mAP is only 0.002.
<img width="459" alt="图片" src="https://user-images.githubusercontent.com/86194554/222046381-e7a28711-4c7b-4581-a682-a4f73ea2018d.png">

**Reply by fredzzhang** [reply]
It is weird that you are getting zero mAP when the IoU threshold is zero. I can't really tell what the problem is. But an easy solution is to start clean. Try and find a new spot and clone the repository again. Make sure you've removed all changes and are on the exact same head as the remote.

Fred.

**Reply by jasminethurder** [reply]
Thank you very much~ I will try that. Really appreciate your reply and the work you have done!!!

**Reply by jasminethurder** [reply]
I have rebuilt the project and removed all changes. But its still 0 mAP when i test on hico-det. Did you test the model in this way? use  `python main.py --eval --resume checkpoints/upt-r50-hicodet.pt` Thanks a lot!

**Reply by jasminethurder** [reply]
I also tried to test using matlab, but the result is also close to 0. May I ask what method you used to test before?

**Reply by fredzzhang** [reply]
Hi @jasminethurder,

You seem to be running the correct commands. If a clean start does not solve the problem, I think most likely there is something wrong with data preparation. First run `git status` and see if there are any changes made, especially in the submodules (untracked content can be ignored). If `git status` does not return anything, then you are on the exact same head as the remote. Make sure the submodule `hicodet` is on head `9dd59c1`. Then maybe do some visualisations on the HICO-DET images, instead of the custom ones and see if you get what you expect.

Fred.

**Reply by jasminethurder** [reply]
It works!!! It should be something wrong with data preparation. Thank you very much for your prompt and friendly reply, which helps me a lot. Appreciate your great work and best wishes to you!
